src/plugins/intel_cpu/src/nodes/concat.cpp

```diff
@@ -164,6 +164,22 @@ void Concat::initSupportedPrimitiveDescriptors() {
         supportedPrimitiveDescriptors.emplace_back(config, impl_desc_type::ref);
         if (itr->first != LayoutType::nspc) {
             pdIndexesToReuse.push_back(supportedPrimitiveDescriptors.size() - 1);
+        } else if (canBeInPlace) {
+            // canBeInPlace means all dims before axis are 1, so for nspc layout we only need check sp dimensions in
+            // axis=1 cases here
+            bool InPlaceCondOk = true;
+            if (axis == 1) {
+                const auto& childDims = outputShapes[0].getDims();
+                for (auto it = childDims.rbegin(); it != childDims.rend() - axis - 1; ++it) {
+                    if (*it != 1) {
+                        InPlaceCondOk = false;
+                        break;
+                    }
+                }
+            }
+            if (InPlaceCondOk) {
```

Comment:
```suggestion
            const auto& childDims = outputShapes[0].getDims();
            if (axis == 1 && std::all_of(childDims.crbegin(), childDims.crend() - 2, [](const Dim dim) {return 1 == dim;} )) {
```

Reply:
Hi, @maxnick : if use this way, does the cases which "axis !=1"  will not be covered? e.g. input shape [1, 1, 32, 1] axis ==2

Reply:
`axis == 1` condition is still there

Reply:
cases( [1, 1, 32, 32] nspc axis = 2) added.

Reply:
Great. So please modify condition `if (axis != 1 || std::all_of(...)) { ...`.

Reply:
done